src/sql/src/plan/statement/ddl.rs

```diff
@@ -2992,27 +3016,98 @@ pub fn plan_create_connector(
         if_not_exists,
     } = stmt;
     let connector_inner = match connector {
-        CreateConnector::Kafka {
-            broker,
-            with_options,
-        } => {
-            let mut with_options = normalize::options(&with_options);
+        CreateConnector::Kafka { broker, security } => {
+            let mut opts_map = BTreeMap::from_iter(
+                security
+                    .iter()
+                    .map(|o| (o.key.to_string().to_uppercase(), o.value.as_ref())),
+            );
+            if opts_map.len() < security.len() {
+                bail!("CONNECTOR configuration option specified more than once");
+            }
+            let security = match opts_map.remove("SECURITY")
+            {
+                Some(Some(WithOptionValue::Value(Value::String(protocol)))) => match protocol.as_str() {
+                    "SSL" => {
+                        let key = resolve_and_track_option(&mut opts_map, "KEY")?;
+                        let certificate = resolve_and_track_option( &mut opts_map, "CERTIFICATE")?;
+                        let passphrase = resolve_and_track_option( &mut opts_map, "PASSWORD")?;
+                        let authority = resolve_and_track_option( &mut opts_map, "AUTHORITY")?;
+                        KafkaSecurityOptions::SSL { key, certificate, passphrase, authority }
+                    }
+                    "SASL-SSL" => {
+                        let mechanism = opts_map.remove("MECHANISM").expect("MECHANISM is expected if SECURITY = SASL-SSL").expect("MECHANISM must have a value if SECURITY is SASL-SSL").to_string();
```

Comment:
These double expects need to be resolved on `opts_map` type, not by calling `expect` twice